stdlib/asm/mem.masm

```diff
@@ -0,0 +1,44 @@
+# ===== MEMORY FUNCTIONS ==========================================================================
+
+#! Copies `n` words from `read_ptr` to `write_ptr`.
+#!
+#! Stack transition looks as follows:
+#! [n, read_ptr, write_ptr, ...] -> [...]
+#! cycles: 15 + 16n
+export.memcopy
+  # The loop variable is changed with an add instead of sub because the former
+  # uses one fewer cycle. So here the counter is negated. (1 cycles)
+  # stack: [-n, read_ptr, write_ptr, ...]
+  neg
+
+  # Pad the stack because mem_load overwrites it (4 cycles)
+  # stack: [PAD, -n, read_ptr, write_ptr, ...]
+  padw
+
+  # check loop condition (3 cycles)
+  # stack: [b, PAD, -n, read_ptr, write_ptr, ...]
+  dup.4 neq.0
+
+  # LOOP: [PAD, -n, read_ptr, write_ptr, ...]
+  # while(n!=0) (16 cycles)
+  while.true
+    # perform read (2 cycles)
+    # stack: [A, -n, read_ptr, write_ptr, ...]
+    dup.5 mem_loadw
+
+    # peform write (2 cycles)
+    # stack: [A, -n, read_ptr, write_ptr, ...]
+    dup.6 mem_storew
+
+    # update counters (9 cycles)
+    swapw # stack: [-n, read_ptr+1, write_ptr+1, x, PAD, ...]
+        add.1 movup.3 movup.3 add.1 movup.3 add.1 movup.3
+    swapw # stack: [PAD, -n, read_ptr+1, write_ptr+1, x, ...]
```

Comment:
Are the comments correct here? Should it be:
```
swapw # stack: [-n, read_ptr, write_ptr, x, PAD, ...]
    add.1 movup.3 movup.3 add.1 movup.3 add.1 movup.3
swapw # stack: [PAD, -n+1, read_ptr+1, write_ptr+1, x, ...]
```

Reply:
also a nit: the middle line doesn't need the extra indentation

Reply:
fixed :)